process-manager.js

```diff
@@ -43,18 +43,14 @@ class ProcessWrapper extends EventEmitter {
 }
 
 class ProcessManager {
-	constructor(options) {
-		if (!options) options = {};
-
+	constructor(execFile, maxProcesses, isChatBased) {
 		this.processes = [];
 		this.taskId = 0;
+		this.execFile = '' + execFile;
+		this.maxProcesses = (!isNaN(maxProcesses) || maxProcesses >= 1) ? maxProcesses : 1;
+		this.isChatBased = !!isChatBased;
 
```

Comment:
I don't see the advantage to using an options object here. Having to know exactly what the names of each of the arguments are to add them to the object from other areas of the code would just complicate refactoring the code if a property name changes. The one upside I could see for it would be making the constructor monomorphic, but at that point you'd have to include all the arguments in the object regardless of if you need them or not. I chose to use arguments instead to make constructing the process manager not require complecting its internal behaviour with the shape of object it requires for an argument.